roles/infrastructure/tasks/teardown_aws_network.yml

```diff
@@ -16,9 +16,7 @@
 
 - name: Tear down AWS Security Groups associated with enviroment and datalake
   when: 
-    - infra__teardown_deletes_network 
     - infra__aws_vpc_id != ""
-    - not infra__force_teardown
```

Comment:
If doing a full teardown, these SGs will get discovered and removed with all others in the VPC

Reply:
1. Security groups are associated with EC2 instances even though strangely they come under VPC. This behavior of SGs getting deleted with instances is consistent with EKS clusters that spin up when we create experiences. https://docs.aws.amazon.com/vpc/latest/userguide/VPC_SecurityGroups.html
2. While creating the infra, we always create the SGs. So we need to delete them when the associated environment is torn down. Else they will remain persistent. We cannot reuse them across environments as each may have its own set of rules. https://github.com/wmudge/cloudera.exe/blob/03e4223e85fb3469a63ee8ebbf32adcb2f1993e0/roles/infrastructure/tasks/setup_aws_network.yml#L89
<img width="938" alt="Screenshot 2021-07-28 at 9 26 40 AM" src="https://user-images.githubusercontent.com/43537803/127263655-15a07634-69c7-4e99-b0ad-cf397f55e25b.png">

Reply:
Ah, yes, I see your point, we must rely on the fact that the entire set of SGs remains whole within the VPC or else face a corrupted set of rules.

So this would mean a slight change in the meaning of the `infra__teardown_deletes_network` in that SGs are now considered part of the EC2 instances and not the network?

Reply:
@wmudge 
Logically, SGs are closer to networks as they are firewall rules applied at network level. However, they get attached to the EC2 instances. Do you think we should move this to compute or in the platform level? 

I believe the current structure is ok. SGs are like the non-permanent pieces of the network layer!